stdlib/universe/map2.go

```diff
@@ -0,0 +1,524 @@
+package universe
+
+import (
+	"context"
+	"sort"
+
+	"github.com/apache/arrow/go/arrow/memory"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/compiler"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/internal/arrowutil"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/internal/mutable"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+//go:generate -command tmpl ../../gotool.sh github.com/benbjohnson/tmpl
+//go:generate tmpl -data=@../../array/types.tmpldata -o map2.gen.go map2.gen.go.tmpl
+
+type mapTransformation2 struct {
+	ctx context.Context
+	fn  mapFunc
+}
+
+func newMapTransformation2(ctx context.Context, id execute.DatasetID, spec *MapProcedureSpec, mem memory.Allocator) (execute.Transformation, execute.Dataset, error) {
+	var fn mapFunc
+	// if spec.Fn.Fn.Vectorized != nil {
+	// 	fn = &mapVectorFunc{
+	// 		fn: execute.NewVectorMapFn(
+	// 			spec.Fn.Fn.Vectorized,
+	// 			compiler.ToScope(spec.Fn.Scope),
+	// 		),
+	// 	}
+	// } else {
```

Comment:
This seems like an important check. Is there a reason we're skipping it?

Reply:
Whoops.